vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/experiments/experiment/ExperimentDetailsComponent.java

```diff
@@ -0,0 +1,233 @@
+  }
+
+  public void setStyles(String... componentStyles) {
+    getContent().addClassNames(componentStyles);
+  }
+
+  private final class Handler {
+
+    private final ProjectInformationService projectInformationService;
+    private final ExperimentInformationService experimentInformationService;
+
+    public Handler(ProjectInformationService projectInformationService,
+        ExperimentInformationService experimentInformationService) {
+      this.projectInformationService = projectInformationService;
+      this.experimentInformationService = experimentInformationService;
+    }
+
+    public void setProjectId(ProjectId projectId) {
+      projectInformationService.find(projectId.value())
+          .ifPresentOrElse(this::getActiveExperimentFromProject, this::emptyAction);
+    }
+
+    private void getActiveExperimentFromProject(Project project) {
+      experimentInformationService.find(project.activeExperiment().value())
+          .ifPresentOrElse(this::loadExperimentInformation, this::emptyAction);
+    }
+
+    private void loadExperimentInformation(Experiment experiment) {
+      getContent().addTitle(experiment.getName());
+      loadTagInformation(experiment);
+      loadSampleOriginInformation(experiment);
+      loadBlockingVariableInformation();
+      experimentalVariableCard.experimentId(experiment.experimentId());
+    }
+
+    private void loadTagInformation(Experiment experiment) {
+      tagLayout.removeAll();
+      experiment.getSpecies().forEach(species -> tagLayout.add(new Span(species.value())));
+      experiment.getSpecimens().forEach(specimen -> tagLayout.add(new Span(specimen.value())));
+      experiment.getAnalytes().forEach(analyte -> tagLayout.add(new Span(analyte.value())));
+      initTagPlusButton();
+    }
+
+    //Todo what should be added here? Additional separate button building and functionality
+    private void initTagPlusButton() {
+      Icon plusIcon = LumoIcon.PLUS.create();
+      plusIcon.addClassNames(IconSize.SMALL);
+      tagLayout.add(plusIcon);
+      tagLayout.getChildren().forEach(component -> {
+        component.getElement().getThemeList().add("badge small");
+        component.getElement().getThemeList().add(FontSize.SMALL);
+      });
+    }
+
+    private void loadSampleOriginInformation(Experiment experiment) {
+      speciesForm.removeAll();
+      specimenForm.removeAll();
+      analyteForm.removeAll();
+      experiment.getSpecies().forEach(species -> speciesForm.add(new Span(species.value())));
+      experiment.getSpecimens().forEach(specimen -> specimenForm.add(new Span(specimen.value())));
+      experiment.getAnalytes().forEach(analyte -> analyteForm.add(new Span(analyte.value())));
+    }
+
+    private void loadBlockingVariableInformation() {
+      //ToDo load information from backend once implemented
+    }
+
+    //ToDo what should happen in the UI if neither project nor experiment has been found?
+    private void emptyAction() {
+
+    }
```

Comment:
I do not think an empty method called `emptyAction` is necessary at the moment. This method only swallows unexpected or undefined behaviour.

Reply:
As you suggested, I removed the empty action 👍